timely_beliefs/beliefs/classes.py

```diff
@@ -1354,13 +1368,28 @@ def resample_events(
                 df = df.reset_index(
                     level=[belief_timing_col, "source", "cumulative_probability"]
                 )
+                resample_ratio = pd.to_timedelta(
+                    to_offset(df.event_resolution)
+                ) / pd.Timedelta(event_resolution)
+                if keep_nan_values:
+                    # back up NaN values
+                    unique_event_value_not_in_df = df["event_value"].abs().sum() + 1
+                    df = df.fillna(unique_event_value_not_in_df)
                 new_index = pd.date_range(
                     start=df.index[0],
-                    periods=len(df) * (self.event_resolution // event_resolution),
+                    end=df.index[-1] + self.event_resolution,
+                    closed="left",
                     freq=event_resolution,
                     name="event_start",
                 )
-                df = df.reindex(new_index).fillna(method="pad")
+                df = df.reindex(new_index).fillna(
+                    method="pad",
+                    limit=math.ceil(resample_ratio) - 1 if resample_ratio > 1 else None,
```

Comment:
Can you comment on this?

Reply:
I thought it was best to explain by example. See the new inline comment.